### ISSUE TITLE
Generate transitions without spin projections

### ISSUE THREAD

**Issue by redeboer** [ComPWA/qrules#219]
### Problem description

QRules is [intended to be a generic system of input (topologies), constraints (initial and final state) and selection rules (conservation laws)](https://qrules.readthedocs.io/en/0.10.x/). Currently it produces specific state transition objects with `Particle` instances on that edges that each have a `float` for spin projections.

These spin projections are however rather a kinematic thing (it depends on rest frame) and there is no quantum conservation rule for these projections alone (in other words, it's not an inherent quantum number that each state caries independent of reference frame).

Some issues related to generating all pin projections:
- It can result in a large number of generated transitions due to combinatorics, making the program slow. See e.g. https://github.com/ComPWA/qrules/issues/27#issuecomment-717359577
- The spin projections are not really needed for AmpForm. In fact, it forms a problem for the implementation of Dalitz-plot decomposition, see https://github.com/ComPWA/ampform/issues/318.

It should be possible to select a different number of input conditions (here: dropping spin projections of the initial and final states). This would allow generating transitions purely with spin magnitudes.

**Comment by redeboer** [comment]
@grayson-helmholz I updated the issue with a sketch of the steps we discussed. Can you create an issue for the first step? Either [the fancy way](https://docs.github.com/en/issues/managing-your-tasks-with-tasklists/managing-tasks-in-a-tasklist#converting-draft-tasks-to-issues-in-a-tasklist) (you should have the rights for that now) or just posting a new issue.

So, just a link to the `python-constraints2` examples, sketch of the example you had in mind, and goal of the issue/task.

**Comment by redeboer** [comment]
Investigation showed this needs an engine-level rework rather than a localized change: spin projections are hard-wired through input expansion, the quantum-number data model, CSP variable creation, rule inputs, settings, output, and IO. That work is now tracked by the epic #336 (which this issue is *blocked by*), where it is broken into staged sub-issues. This issue stays as the motivating deliverable and can be closed once the epic delivers a projection-free generation path.
